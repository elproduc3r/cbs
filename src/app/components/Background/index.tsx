"use client"

import React, {createContext, useState, MouseEventHandler} from 'react';
import styled from "styled-components";
import getRandomColors from '../../utils/getRandomColors';

export interface BackgroundProps {
  children: React.ReactNode;
};

interface BackgroundContextType {
  changeBackground: MouseEventHandler<HTMLButtonElement>;
}

export const BackgroundContext = createContext<BackgroundContextType>({
  changeBackground: () => {}
});

interface StyledBackgroundProps {
  colors: string[];
};

const StyledBackground = styled.div<StyledBackgroundProps>`
  background-image: linear-gradient(to bottom right, 
    ${(props:StyledBackgroundProps) => props.colors[0]}, ${(props:StyledBackgroundProps) => props.colors[1]});
  height: 100vh;
`;

const Background = ({
  children
}: BackgroundProps):JSX.Element => {

  const [colors, setColors] = useState(["#1d6dd5", "#d529cc"]);
  const changeBackground = (): void => {

    const newColors = getRandomColors(2);
    setColors(newColors);
  };

  return (
    <BackgroundContext.Provider value={{changeBackground}}>
      <StyledBackground colors={colors}>
        {children}
      </StyledBackground>
    </BackgroundContext.Provider>
  );
};

export default Background;
