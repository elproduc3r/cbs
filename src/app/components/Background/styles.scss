@keyframes fadeIn {
  0%   {  opacity: 0%; }
  100% {  background-size: 100%; }      
 }

.background-gradient {
  transition: background 1s;
  height: 100vh;
  width: 100%;
  max-width: 100%;
  overflow: auto;
  &.animation {
    animation: fadeIn 1s ease-in 1;
  }
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 1;
}

.background-container {
  height: 100vh;
  width: 100%;
  overflow: auto;
  z-index: 2;
  position: absolute;
}

