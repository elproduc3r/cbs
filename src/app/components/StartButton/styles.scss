.start-button {
  padding: 10px 42px;
  background-color: #ffffff;
  border-radius: 28px;
  font-weight: 800;
  font-family: inherit;
  font-size: 1.5rem;
  border-width: 0px;
  align-self: flex-start;
  text-transform: uppercase;
  transition: .5s;
  margin-top: 2rem;
  &:hover {
    color: forestgreen;
  }
}

