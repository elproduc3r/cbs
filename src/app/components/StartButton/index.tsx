"use client"

import { useContext } from "react";
import { BackgroundContext } from "../Background";

const StartButton = () => {

  const {changeBackground} = useContext(BackgroundContext);
  return (
    <button
      onClick={changeBackground}
    >
      Get Started
    </button>
  )
}

export default StartButton;