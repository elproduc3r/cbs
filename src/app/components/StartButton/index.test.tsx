import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import {BackgroundContext} from '../Background';
import StartButton from '../StartButton';

test('StartButton snapshot', () => {
  const {container} = render(
    <StartButton />
  );
  expect(container).toMatchSnapshot();
});

test('onClick calls function from BackgroundContext.Provider', () => {
  const mockFn = jest.fn(():void => {});
  render(
    <BackgroundContext.Provider value={{changeBackground: mockFn}}>
      <StartButton />
    </BackgroundContext.Provider>
  );
  const button = screen.getByRole("button");
  fireEvent.click(button);
  expect(mockFn).toHaveBeenCalled();
});
