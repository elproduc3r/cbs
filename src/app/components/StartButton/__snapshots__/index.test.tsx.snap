// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartButton snapshot 1`] = `
<div>
  <button
    class="start-button"
  >
    Get Started
  </button>
</div>
`;
